pkg/auth/config.go

```diff
@@ -81,9 +81,10 @@ type Config struct {
 	//Config only for AKS
 	NodeResourceGroup string `json:"nodeResourceGroup" yaml:"nodeResourceGroup"`
 	//SubnetId is the resource ID of the subnet that VM network interfaces should use
-	SubnetID   string `json:"subnetId" yaml:"subnetId"`
-	VnetName   string `json:"vnetName" yaml:"vnetName"`
-	SubnetName string `json:"subnetName" yaml:"subnetName"`
+	SubnetID          string `json:"subnetId" yaml:"subnetId"`
+	VnetName          string `json:"vnetName" yaml:"vnetName"`
+	VnetResourceGroup string `json:"vnetResourceGroup" yaml:"vnetResourceGroup"`
```

Comment:
was thinking this too. Lets do this

Reply:
this becomes a bit more complex of a refactor than i initially thought because in bootstrap.go we set the vnet labels. 


But in my subnet per nodeclass pr, I already do the refactoring required to make this easy.  @tallaxes  would you mind if i merged the two prs? Since so much of the refactoring for VnetName and SubnetName happens in that pr?

Reply:
Given the nuances of this problem space, I would very much prefer to iron out the single subnet version first, without mixing it with per-nodepool. Would it be possible to apply / bring over the required refactoring without mixing the two yet? (Cherry-picking maybe?)

Reply:
Its doable via copy paste just a bit annoying since I already did the work over on the other pr but i can do that as i agree adding vnet resource group just to remove it one pr later doesn't make sense

Reply:
ok did the entire refactor on this pr only took 50 minutes. Not easily cherry pickable since different parts reference the nodeclass vnetSubnetID. Let me know what you think